File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();

        int mini = prices[0] , maxProfit = INT_MIN;

        for(int i = 1 ; i < n ; i++){
            maxProfit = max(prices[i]-mini , maxProfit);

            mini = min(mini , prices[i]);
        }

        return maxProfit > 0 ? maxProfit : 0;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int n = prices.Length;
        // [7,1,5,3,6,4]   
        int ans = 0;
        int mini = prices[0];

        for(int i = 0 ; i < n ; i++){
            ans = Math.Max(ans, prices[i]-mini); // 4, 5 
            mini = Math.Min(mini, prices[i]);
        }

        return ans;
    }
}